### ISSUE TITLE
Duplicate contacts problem

### ISSUE THREAD

**Issue by chrislim** [ceaseless-prayer/CeaselessIOS#160]
This is a serious issue. Contacts are showing up multiple times over time as they are syncing.

**Comment by chrislim** [comment]
#155

**Comment by chrislim** [comment]
I wonder if this happens when I delete contacts, which causes a change in local ID
The next sync run might fail to lookup by id, fail to lookup by first and last name and so create a new contact.

Alternatively, it could be that a new contact is created as part of the regular process, but somewhere along the way, the old contact does not get cleaned up.

We may benefit from a hash comparison approach for faster lookups too...instead of recreating contacts every time, just check if the hash is the same.

**Comment by chrislim** [comment]
The other related question is how to help users who are already encountering this duplicate contact problem. Do we clean it up for them? If we find two contacts with exactly the same information?

**Comment by LIL717** [comment]
Here is a specific case of this:  In Contacts I have a friend Jill Allison.  I see her once in my Address Book, but a look at her record reveals that there are two occurrences of her.  The "iPhone" version has a photo, her name and a mailing address.  The "iCloud" version has no information for her other than a photo.  
In Ceaseless there are two entries for her.  The first does not have a photo, the second does have a photo.  View in Contacts for either of these entries shows the contact as I see it in the Address book with a photo and an address.

**Comment by LIL717** [comment]
The case I described above does not happen consistently.  I have deleted the app, built from Xcode, then   downloaded from app store.  Not seeing the dups for those with only first names or for the contact, Jill Allison mentioned above.  I have also tried deleting the app, then downloading from the app store, manually syncing - no dups.   I have also tried deleting the app, then downloading from the app store, switching into dev mode closing and opening several times - no dups.

.  .
